### ISSUE TITLE
Impl BatchNorm

### ISSUE THREAD

**Issue by drahnr** [ratpoison-io/collenchyma#11]
Starting points:
https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/batch_norm_layer.html
Google Paper:
https://arxiv.org/abs/1502.03167
